src/formatDate.js

```diff
@@ -50,7 +50,37 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+  const temporaryArray = [...splitDate];
+  const newArray = [];
+  const yearFromIndex = fromFormat.findIndex(item => item.includes('Y'));
+  const yearToIndex = toFormat.findIndex(item => item.includes('Y'));
+
+  if (fromFormat[yearFromIndex].length !== toFormat[yearToIndex].length) {
+    temporaryArray[yearFromIndex] = convertYear(temporaryArray[yearFromIndex]);
+
+    fromFormat[yearFromIndex] = toFormat[yearToIndex];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const index = toFormat.indexOf(fromFormat[i]);
+
+    newArray[index] = temporaryArray[i];
+  }
+
+  return newArray.join(toFormat[toFormat.length - 1]);
+}
+
+function convertYear(str) {
```

Comment:
```suggestion
function convertYear(year) {
```
let's use appropriate names for variables